GoogleForCreators/web-stories-wp
# Floating Menu: Allow fixing / disabling the element menu

## Context


## Summary
Adds a new Settings menu to the Element toolbar/floating menu that allows moving the menu to the top / to the element / allows persistently closing the toolbar.


## Relevant Technical Choices
- Increases the header height a bit to fit the toolbar


## To-do

- [x] Confirm UX
- [ ] Add icon from UX / update copy



## User-facing changes
<img width="603" alt="Screenshot 2022-08-17 at 17 38 58" src="https://user-images.githubusercontent.com/3294597/185165640-dcb06a05-e43c-4c83-9de8-7b1db00a3145.png">
<img width="948" alt="Screenshot 2022-08-17 at 17 38 51" src="https://user-images.githubusercontent.com/3294597/185165661-cfa716e0-f490-4e64-a959-e9208ee54499.png">




## Testing Instructions






This PR can be tested by following these steps:

1. Add an element
2. See the new Settings menu in the floating menu.
3. Verify that it's possible to place the element above the page using the "Fix to top" option and back to the element with the "Fix to element" option.
4. Verify that closing the menu from "Always hide" will persistently hide the menu (deselecting and selecting the element again will not reopen the menu)
5. Verify that when closing the menu from "Always hide", a new button appears to the bottom right area and that button is highlighted after closing the menu.
6. Verify that the button opens the menu again


## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
```
trackEvent('floating_menu', {
      name: 'reposition_menu',
      position,
    });

```
```
trackEvent('floating_menu', {
      name: 'persistent_menu_hide',
    });
```


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #11570
